soc/riscv/riscv-ite/it8xxx2/linker.ld

```diff
@@ -147,6 +151,15 @@ SECTIONS
 		. = ALIGN(0x1000);
 		/* Mapping base address must be 4k-aligned */
 		__ilm_flash_start = .;
+#ifdef CONFIG_SOC_IT8XXX2_SHA256_HW_ACCELERATE
+		/* Dummy section for sha256 */
+		__sha256_pseudo_block_start = .;
+		KEEP(*(.__sha256_pseudo_block))
```

Comment:
You don't have any code using the linker section `__sha256_pseudo_block`. Is this PR missing a file?

As implemented, I don't see how the location counter will move forward, which will then cause the `ASSERT` at line 160 to file.

Reply:
This section doesn't actually need to contain any data, right? It seems to only exist to ensure the flash and RAM addresses for ILM keep the same 4k-relative alignment, so this can skip taking any sections and simply advance the location counter by the appropriate amount- then your code doesn't need an unused array either.

Reply:
First 4kB of ram is configured as ILM, if we don't declare __sha256_pseudo_block, then this 512B block will be put text code, so we need to declare __sha256_pseudo_block to avoid text section to be re-direct to ram.

Reply:
Yes, but there doesn't need to be any data actually placed here:

```
#ifdef CONFIG_SOC_IT8XXX2_SHA256_HW_ACCELERATE
    /* Pad to match allocation of block in RAM, maintaining code alignment against ILM */
    . = . + SHA256_BLOCK_SIZE;
#endif
```

There's no data of use in flash, so there don't need to be any symbols or sections.

Reply:
Done